StayFit/CommunityViewController.m

```diff
@@ -0,0 +1,221 @@
+//
+//  CommunityViewController.m
+//  StayFit
+//
+//  Created by Nikita Agarwal on 7/11/22.
+//
+
+#import "CommunityViewController.h"
+#import "Parse/Parse.h"
+#import "FitnessFeedCell.h"
+#import "recipesPost.h"
+#import "NewPostViewController.h"
+#import "RecipesFeedCell.h"
+#import "GymBuddyCell.h"
+#import <QuartzCore/QuartzCore.h>
+@interface CommunityViewController ()<UITableViewDataSource, UITableViewDelegate>
+@property (strong, nonatomic) NSArray *arrayOfFitnessPosts;
+@property (strong, nonatomic) NSArray *arrayofRecipesPosts;
+@property (strong, nonatomic) NSArray *arrayOfGymBuddies;
+@property (strong, nonatomic) UIRefreshControl *refreshControl;
+@end
+
+@implementation CommunityViewController
+@synthesize segout;
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    //refreshes the page each time a post is added
+    self.refreshControl = [[UIRefreshControl alloc] init];
+    [self.refreshControl addTarget:self action:@selector(fetchPosts) forControlEvents:UIControlEventValueChanged];
+    [self.fitnessTableView insertSubview:self.refreshControl atIndex:0];
+    self.fitnessTableView.dataSource = self;
+    self.fitnessTableView.delegate = self;
+    [self fitnessCase];
+    
+}
+
+- (IBAction)segact:(id)sender {
+    //checks which segment is active
+    switch (self.segout.selectedSegmentIndex){
+        case 0:
+            [self fitnessCase];
+            break;
+        case 1:
+            [self recipesCase];
+            break;
+        case 2:
+            [self gymBuddy];
+            break;
+    }
+}
+
+-(void) fitnessCase{
+    self.addButton.hidden = false;
+    self.arrayOfFitnessPosts = [[NSMutableArray alloc] init];
+    [self fetchFitnessPosts];
+}
+
+-(void) recipesCase{
+    self.addButton.hidden = false;
+    self.arrayofRecipesPosts = [[NSMutableArray alloc] init];
+    [self fetchRecipesPosts];
+}
+
+-(void) gymBuddy{
+    self.addButton.hidden = true;
+    self.arrayOfGymBuddies = [[NSMutableArray alloc] init];
+    [self fetchGymBuddies];
+}
+
+-(void) fetchPosts {
+    if (self.segout.selectedSegmentIndex == 0) {
+        [self fetchFitnessPosts];
+    } else if (self.segout.selectedSegmentIndex == 1) {
+        [self fetchRecipesPosts];
+    } else {
+        [self fetchGymBuddies];
+    }
+}
+
+-(void) fetchFitnessPosts{
+    //creates a new query for fitness posts
+        PFQuery *postQuery = [PFQuery queryWithClassName:@"Post"];
+        [postQuery orderByDescending:@"createdAt"];
+        [postQuery includeKey:@"author"];
+        postQuery.limit = 20;
+    
+        [postQuery findObjectsInBackgroundWithBlock:^(NSArray *posts, NSError *error) {
+            if (posts) {
+                self.arrayOfFitnessPosts = posts;
+                [self.fitnessTableView reloadData];
+            }
+            else {
+                NSLog(@"%@", error);
```

Comment:
might be good to log more specific error information here (i.e. what kind of error is it, what function did it occur in)